test/acceptance/service_cluster_ip_test.go

```diff
@@ -0,0 +1,76 @@
+// +build acceptance
+
+package acceptance
+
+import (
+	"encoding/json"
+	"testing"
+
+	tfstatehelper "github.com/hashicorp/terraform-provider-kubernetes-alpha/test/helper/state"
+)
+
+// This test case tests a Service but also is a demonstration of some the assert functions
+// available in the test helper
+func TestKubernetesManifest_Service_ClusterIP(t *testing.T) {
+	name := randName()
+	namespace := randName()
+
+	tf := tfhelper.RequireNewWorkingDir(t)
+	tf.SetReattachInfo(reattachInfo)
+	defer func() {
+		tf.RequireDestroy(t)
+		tf.Close()
+		k8shelper.AssertNamespacedResourceDoesNotExist(t, "v1", "services", namespace, name)
+	}()
+
+	k8shelper.CreateNamespace(t, namespace)
+	defer k8shelper.DeleteNamespace(t, namespace)
+
+	tfvars := TFVARS{
+		"namespace": namespace,
+		"name":      name,
+	}
+	tfconfig := loadTerraformConfig(t, "Service_ClusterIP/service.tf", tfvars)
+	tf.RequireSetConfig(t, tfconfig)
+	tf.RequireInit(t)
+	tf.RequireApply(t)
+
+	k8shelper.AssertNamespacedResourceExists(t, "v1", "services", namespace, name)
+
+	tfstate := tfstatehelper.NewHelper(tf.RequireState(t))
+	tfstate.AssertAttributeValues(t, tfstatehelper.AttributeValues{
+		"kubernetes_manifest.test.object.metadata.namespace":      namespace,
+		"kubernetes_manifest.test.object.metadata.name":           name,
+		"kubernetes_manifest.test.object.spec.ports.0.name":       "http",
+		"kubernetes_manifest.test.object.spec.ports.0.port":       json.Number("80"),
+		"kubernetes_manifest.test.object.spec.ports.0.targetPort": json.Number("8080"),
+		"kubernetes_manifest.test.object.spec.ports.0.protocol":   "TCP",
+		"kubernetes_manifest.test.object.spec.selector.app":       "test",
+		"kubernetes_manifest.test.object.spec.type":               "ClusterIP",
```

Comment:
When a service of type "ClusterIP" is created, the resulting resource gets two attributes added by the controller manager which hold the actual cluster IP address(es) allocated to the service. These were causing inconsistent state issues in the pre-0.3.0 provider and are a simple demonstration of why we need the OpenAPI type system.

It would be useful to add assertions for the presence of these two attributes which demonstrate the resource is correctly typed and Terraform saves these computed values as well.

Here an example of how they show up in state:
```
resource "kubernetes_manifest" "test" {
    manifest = {
        apiVersion = "v1"
        kind       = "Service"
        metadata   = {
            name      = "alextest"
            namespace = "default"
        }
        spec       = {
            ports    = [
                {
                    name       = "http"
                    port       = 80
                    protocol   = "TCP"
                    targetPort = 8080
                },
            ]
            selector = {
                app = "test"
            }
            type     = "ClusterIP"
        }
    }
    object   = {
        apiVersion = "v1"
        kind       = "Service"
        metadata   = {
            annotations                = null
            clusterName                = null
            creationTimestamp          = null
            deletionGracePeriodSeconds = null
            deletionTimestamp          = null
            finalizers                 = null
            generateName               = null
            generation                 = null
            labels                     = null
            managedFields              = null
            name                       = "alextest"
            namespace                  = "default"
            ownerReferences            = null
            resourceVersion            = null
            selfLink                   = null
            uid                        = null
        }
        spec       = {
            allocateLoadBalancerNodePorts = null
            clusterIP                     = "10.104.36.74"
            clusterIPs                    = [
                "10.104.36.74",
            ]
            externalIPs                   = null
            externalName                  = null
            externalTrafficPolicy         = null
            healthCheckNodePort           = null
            ipFamilies                    = null
            ipFamilyPolicy                = null
            loadBalancerIP                = null
            loadBalancerSourceRanges      = null
            ports                         = [
                {
                    appProtocol = null
                    name        = "http"
                    nodePort    = null
                    port        = 80
                    protocol    = "TCP"
                    targetPort  = 8080
                },
            ]
            publishNotReadyAddresses      = null
            selector                      = {
                "app" = "test"
            }
            sessionAffinity               = "None"
            sessionAffinityConfig         = {
                clientIP = {
                    timeoutSeconds = null
                }
            }
            topologyKeys                  = null
            type                          = "ClusterIP"
        }
    }
}
```